### PR TITLE
Switch tests from zemu to ragger

### PR DESCRIPTION
**Pull request by tdejoigny-ledger**
First PR to initiate the transition from Zemu framework to Ragger framework.

As soon as this [PR ](https://github.com/LedgerHQ/app-ethereum/pull/449) is merged, we can :
- remove the Ethereum client from the repo
- update the test requirements file "requirement.txt" to use directly the eth client available on pypi
- update the swap test (and the CI) in order to build a known tx payload (instead of using a raw transaction from Etherscan) and to sign it (allowing the removal of flag "BYPASS_SIGNATURES=1" in ETH app)
- use the correct branch for plugin-sdk sub-module (>>develop branch)

### CODE REVIEW

File: .github/workflows/build_and_functional_tests.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    needs:
+      - build_plugin
+      - build_ethereum_app
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_ragger_tests.yml@v1
+    with:
+      app_repository: LedgerHQ/app-plugin-boilerplate
+      download_app_binaries_artifact: ethereum_binaries
+      lib_binaries_artifact: libraries_binaries
+      test_dir: tests
+      run_for_devices: '["nanos", "nanox", "nanosp", "stax"]'
```
Comment (fbeutin-ledger) [inline .github/workflows/build_and_functional_tests.yml]: new line at end of files

Comment (fbeutin-ledger) [reply .github/workflows/build_and_functional_tests.yml]: IDE can/should do it for you ! :)

File: .github/workflows/build_and_functional_tests.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+jobs:
+  build_plugin:
+    name: Build plugin using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: LedgerHQ/app-plugin-boilerplate
+      flags: "DEBUG=1"
+      upload_app_binaries_artifact: libraries_binaries
+      upload_as_lib_artifact: PluginBoilerplate
+      run_for_devices: '["nanos", "nanox", "nanosp", "stax"]'
```
Comment (fbeutin-ledger) [inline .github/workflows/build_and_functional_tests.yml]: To remove, no param == all

File: .github/workflows/build_and_functional_tests.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  build_ethereum_app:
+    name: Build Ethereum app using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: LedgerHQ/app-ethereum
+      app_branch_name: develop
+      flags: "DEBUG=1 BYPASS_SIGNATURES=1"
+      upload_app_binaries_artifact: ethereum_binaries
+      run_for_devices: '["nanos", "nanox", "nanosp", "stax"]'
```
Comment (fbeutin-ledger) [inline .github/workflows/build_and_functional_tests.yml]: To remove, no param == all

File: .github/workflows/build_and_functional_tests.yml (hunk 1 above)
Comment (fbeutin-ledger) [inline .github/workflows/build_and_functional_tests.yml]: To remove, no param == all

File: .github/workflows/build_and_functional_tests.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  ragger_tests:
+    name: Run ragger tests using the reusable workflow
+    needs:
+      - build_plugin
+      - build_ethereum_app
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_ragger_tests.yml@v1
+    with:
+      app_repository: LedgerHQ/app-plugin-boilerplate
```
Comment (fbeutin-ledger) [inline .github/workflows/build_and_functional_tests.yml]: To remove, no param == self

File: .github/workflows/build_and_functional_tests.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      upload_app_binaries_artifact: libraries_binaries
+      upload_as_lib_artifact: PluginBoilerplate
+      run_for_devices: '["nanos", "nanox", "nanosp", "stax"]'
+
+  build_ethereum_app:
+    name: Build Ethereum app using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: LedgerHQ/app-ethereum
+      app_branch_name: develop
```
Comment (fbeutin-ledger) [inline .github/workflows/build_and_functional_tests.yml:34]: develop or master ? I think it's important to test both, we might want to add the two

File: .github/workflows/build_and_functional_tests.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      - main
+      - develop
+  pull_request:
+
+jobs:
+  build_plugin:
+    name: Build plugin using the reusable workflow
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: LedgerHQ/app-plugin-boilerplate
```
Comment (fbeutin-ledger) [inline .github/workflows/build_and_functional_tests.yml]: To remove, no param == self

File: .gitignore [hunk 7]
```diff
@@ -19,7 +20,11 @@ __pycache__/
...
-tests/lib
-tests/elfs
+# Virtual env for sideload (macOS and Windows)
+ledger/
+
+# Temporary directory with snapshots taken during test runs
+tests/snapshots-tmp/
+
+# Temporary directory with all the plugin binairies (for each device)
+tests/lib_binaries
```
Comment (fbeutin-ledger) [inline .gitignore]: new line at eof

Comment (fbeutin-ledger) [inline .gitignore]: On exchange I have created a token file inside to have github create the directory
https://github.com/LedgerHQ/app-exchange/blob/develop/test/python/lib_binaries/.compiled_sideloaded_applications_go_there

Then here
https://github.com/LedgerHQ/app-exchange/blob/develop/.gitignore#L44
we ignore all `elf` and `zip`

File: tests/README.md [hunk 8]
```diff
@@ -1,45 +1,53 @@
...
+`build/` directory, and named as follow:
 
-To use this script, you will need the
-[Ethereum application](https://github.com/LedgerHQ/app-ethereum) to be cloned
-somewhere, and the path of its repository exported in the `APP_ETHEREUM`
-environment variable, or updated in the script, or as an argument of the script.
+* `build/nanos/bin/app.elf`
+* `build/nanos2/bin/app.elf`
+* `build/nanox/bin/app.elf`
+* `build/stax/bin/app.elf`
```
Comment (fbeutin-ledger) [inline tests/README.md:20]: It means that the direct compilation output of the app is not used as-is for testing ? We need to copy the boilerplate plugin to the subdir AND copy the eth files to the boilerplate build directory ? Where they will be overwritten if we compile again ?

Comment (tdejoigny-ledger) [reply tests/README.md:20]: Exactly ;-)
this is one of the improvement we need to do, be able to run easily the test locally

File: .github/workflows/build_and_functional_tests.yml (hunk 5 above)
Comment (tdejoigny-ledger) [reply .github/workflows/build_and_functional_tests.yml:34]: yes it is a good question !
I guess it will depend on the plugin branch (develop or master)

Comment (fbeutin-ledger) [reply .github/workflows/build_and_functional_tests.yml:34]: I think it does not depend, when we develop a new version of the plugin (develop branch), we want it to be compatible with the currently deployed ethereum (master branch) and the next deployed ethereum (develop branch)
We should add both, I think

Comment (fbeutin-ledger) [reply .github/workflows/build_and_functional_tests.yml:34]: The CI will never really run on the master branch of the plugin

File: tests/README.md (hunk 8 above)
Comment (fbeutin-ledger) [reply tests/README.md:20]: Ok, I'll try to solve this by modifying the local ragger conftest of the plugin

**Comment by fbeutin-ledger** [comment]
https://github.com/LedgerHQ/app-plugin-boilerplate/pull/105

**Pull request closed without merging**